tests/test_main.cpp

```diff
@@ -10,8 +10,43 @@
 #include "player.h"
 #include "worldfactory.h"
 #include "debug.h"
+#include "mod_manager.h"
 
-void init_global_game_state() {
+std::vector<std::string> extract_mod_selection( std::vector<const char *> &arg_vec )
+{
+    std::vector<std::string> ret;
+    static const char *mod_tag = "--mods=";
+    std::string mod_string;
+    for( auto iter = arg_vec.begin(); iter != arg_vec.end(); iter++ ) {
+        if( strncmp( *iter, mod_tag, strlen( mod_tag ) ) == 0 ) {
+            mod_string = std::string( &(*iter)[ strlen( mod_tag ) ] );
+            arg_vec.erase( iter );
+            break;
+        }
+    }
+
+    const char delim = ',';
+    size_t i = 0;
+    size_t pos = mod_string.find( delim );
+    if( pos == std::string::npos && !mod_string.empty() ) {
+        ret.push_back( mod_string );
+    }
+
+    while( pos != std::string::npos ) {
```

Comment:
Btw. there is a `auto_pickup::split` function in src/auto_pickup.cpp, which seems to do pretty much this algorithm (but via a stringstream, probably not very efficient). It might be good to merge the code here and that function into a proper utility function (maybe in cata_utility.cpp).

But that would be more of a long-term goal, not necessary for this PR.